pkg/cmd/roachtest/spec/cluster_spec.go

```diff
@@ -444,26 +463,66 @@ func (s *ClusterSpec) RoachprodOpts(
 		}
 	}
 
+	// If WorkloadNodeCPUs is unspecified, then we attempt to assign it a smaller CPU VM.
+	// The workload node is generally not as stressed as a crdb node, and potentially giving
+	// it 8-32 CPUs like the rest of the cluster may be overkill.
+	//
+	// N.B. if you change the logic here, make sure to change it in TotalCPUs as well,
+	// which is used to determine CPU quota allocation.
+	if s.WorkloadNodeCPUs == 0 {
+		// If WorkloadNodeCPUs is not specified, then default to a VM with 4 CPUs, unless the
+		// CRDB nodes are already lower than that. In which case, keep the CPU count the same.
+		s.WorkloadNodeCPUs = min(s.CPUs, DefaultWorkloadNodeCPUs)
+
+		// In the case where the crdb cluster is not very small, we may need the higher CPU
```

Comment:
Nit: dunno what "very small" really means; "small" should suffice

Reply:
Dropped this commit.